### PR TITLE
feat: add timeouts configuration

### PR DESCRIPTION
**Pull request by Lisenish**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
+
 const UniqueIndexViolationErrCode = '23505'
-const pool = new Pool({ connectionString: process.env.DATABASE_URL })
+
+const DefaultTimeout = 20_000
+const DefaultPoolSize = 10
+
+const pool = new Pool({
+  connectionString: process.env.DATABASE_URL,
+  connectionTimeoutMillis: Number(process.env.DATABASE_CONN_TIMEOUT ?? DefaultTimeout),
```
Comment (Valve) [inline src/index.ts]: CONN is not a typical acronym, I suggest to make it full (CONNECTION).

File: src/index.ts [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
+  query_timeout: Number(process.env.DATABASE_QUERY_TIMEOUT ?? DefaultTimeout),
```
Comment (Valve) [inline src/index.ts]: is kebab case correct here? I know PG is using it, but is the node driver passing through or camelizing it?

File: src/index.ts [hunk 3]
```diff
@@ -1,7 +1,19 @@
...
+  idle_in_transaction_session_timeout: Number(process.env.DATABASE_IDLE_TRAN_TIMEOUT ?? DefaultTimeout),
```
Comment (Valve) [inline src/index.ts]: TRAN should be TRANSACTION

File: src/index.ts (hunk 2 above)
Comment (Valve) [reply src/index.ts]: default timeout shouldn't be the default for both dialing a connection and executing a query, the latter is usually orders of magnitude longer.

File: src/index.ts [hunk 4]
```diff
@@ -1,7 +1,19 @@
 import { Pool, PoolClient } from 'pg'
 import { Migrator } from './migrator'
+
 const UniqueIndexViolationErrCode = '23505'
-const pool = new Pool({ connectionString: process.env.DATABASE_URL })
+
+const DefaultTimeout = 20_000
```
Comment (Valve) [inline src/index.ts]: this is probably too coarse.

**Review by Valve — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (Lisenish) [reply src/index.ts]: Fixed

File: src/index.ts (hunk 2 above)
Comment (Lisenish) [reply src/index.ts]: >is kebab case correct here

Yes it's kebab here (actually it's snake case), for some reason they have such inconsistency in param names, they probably left original params unchanged in name but those which they somehow mapping they named in camelCase.

>default timeout shouldn't be the default for both dialing a connection and executing a query, the latter is usually orders of magnitude longer.

Yeah I know, decided to keep it simple here, but you're right, fixed.

File: src/index.ts (hunk 3 above)
Comment (Lisenish) [reply src/index.ts]: Fixed

File: src/index.ts (hunk 4 above)
Comment (Lisenish) [reply src/index.ts]: Fixed

**Comment by Valve** [comment]
@Lisenish will you release to NPM?

**Pull request merged**